### PR TITLE
Shayan/60548/Test Tools Integration

### PR DESCRIPTION
**Pull request by iman-fs**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by prince-deriv — APPROVED** [review]

File: __mocks__/gatsby.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            ...rest
+        }) =>
+            React.createElement('a', {
+                ...rest,
+                href: to,
+            }),
+    ),
+    StaticQuery: jest.fn(),
+    useStaticQuery: jest.fn(),
+}
```
Comment (ali-hosseini-fs) [inline __mocks__/gatsby.js:27]: no new line at the end of the file. please add it

File: loadershim.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+global.___loader = {
+    enqueue: jest.fn(),
+}
```
Comment (ali-hosseini-fs) [inline loadershim.js:3]: no new line at the end of file

File: package.json [hunk 3]
```diff
@@ -113,6 +126,9 @@
         "start-80": "sudo gatsby develop -p 80",
         "serve": "gatsby serve",
         "test": "npm run eslint && npm run stylelint",
+        "jest": "jest --passWithNoTests --detectOpenHandles",
+        "jest:coverage": "npm run jest -- --coverage",
+        "jest:watch": "npm run jest -- --watch",
```
Comment (ali-hosseini-fs) [inline package.json:131]: what is that extra `--` in this script?

File: setupTests.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import '@testing-library/jest-dom'
+import '@testing-library/jest-dom/extend-expect'
+import 'jest-extended'
+import 'jest-chain'
```
Comment (ali-hosseini-fs) [inline setupTests.js:4]: it needs new line at the end

File: svgTransform.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// this file helps jest to load svg assets
+
+module.exports = {
+    process() {
+        return 'module.exports = {};'
+    },
+    getCacheKey() {
+        return 'svgTransform'
+    },
+}
```
Comment (ali-hosseini-fs) [inline svgTransform.js:10]: please add a new line in the end

**Review by ali-hosseini-fs — COMMENTED** [review]
left some comments

File: package.json (hunk 3 above)
Comment (iman-fs) [reply package.json:131]: @ali-hosseini-fs It's for passing arguments. 
please check this link: https://docs.npmjs.com/cli/v8/commands/npm-run-script

Comment (iman-fs) [reply package.json:131]: without it the coverage won't work. I did some search and according to this [link](https://stackoverflow.com/questions/24825860/how-to-get-the-code-coverage-report-using-jest) I found the issue.

**Review by michio-rework — DISMISSED** [review]

**Review by ali-hosseini-fs — DISMISSED** [review]

**Review by prince-deriv — DISMISSED** [review]

**Review by michio-rework — APPROVED** [review]

**Review by mohammad-rework — APPROVED** [review]

**Pull request merged**